integration/src/integration/java/pl/allegro/tech/hermes/integration/env/FrontendStarter.java

```diff
@@ -28,79 +30,86 @@
+        args.add(getArgument(FRONTEND_THROUGHPUT_TYPE, "fixed"));
+        args.add(getArgument(FRONTEND_THROUGHPUT_FIXED_MAX, 50 * 1024L));
+        args.add(getArgument(FRONTEND_GRACEFUL_SHUTDOWN_ENABLED, false));
+        args.add(getArgument(FRONTEND_MESSAGE_PREVIEW_ENABLED, true));
+        args.add(getArgument(FRONTEND_MESSAGE_PREVIEW_LOG_PERSIST_PERIOD, 1));
+        return args;
     }
 
     @Override
     public void start() throws Exception {
         LOGGER.info("Starting Hermes Frontend");
-        hermesFrontend = HermesFrontend.frontend()
-            .withBinding(configFactory, ConfigFactory.class)
-            .withHeadersPropagator(new TraceHeadersPropagator())
-            .withLogRepository(serviceLocator -> new MongoLogRepository(FongoFactory.hermesDB(),
-                    10,
-                    1000,
-                    configFactory.getStringProperty(Configs.KAFKA_CLUSTER_NAME),
-                    configFactory.getStringProperty(Configs.HOSTNAME),
-                    serviceLocator.getService(MetricRegistry.class),
-                    serviceLocator.getService(PathsCompiler.class)))
-            .withKafkaTopicsNamesMapper(
-                    new IntegrationTestKafkaNamesMapperFactory(configFactory.getStringProperty(Configs.KAFKA_NAMESPACE)).create())
-            .withDisabledGlobalShutdownHook()
-            .withDisabledFlushLogsShutdownHook()
-            .build();
-
         client = new OkHttpClient();
-        hermesFrontend.start();
+        setSpringProfilesArg();
+        applicationContext = application.run(args.toArray(new String[0]));
         waitForStartup();
     }
 
     @Override
     public void stop() throws Exception {
         LOGGER.info("Stopping Hermes Frontend");
-        hermesFrontend.stop();
+        instance().close();
     }
 
     @Override
-    public HermesFrontend instance() {
-        return hermesFrontend;
+    public ConfigurableApplicationContext instance() {
+        return applicationContext;
     }
 
-    public ConfigFactory config() {
-        return configFactory;
+    public void overrideProperty(Configs config, Object value) {
+        args.add(getArgument(config, value));
     }
 
-    public void overrideProperty(Configs config, Object value) {
-        configFactory.overrideProperty(config, value);
+    public void addSpringProfiles(String... profiles) {
+        String profilesString = Arrays.stream(profiles).collect(Collectors.joining(",", "", ""));
+        this.profiles.add(profilesString);
     }
 
-    private void waitForStartup() throws Exception {
+    private void setSpringProfilesArg() {
+        String profilesString = profiles.stream().collect(Collectors.joining(",", "", ""));
+        args.add("--spring.profiles.active=" + profilesString);
+    }
 
+    private void waitForStartup() {
         await().atMost(adjust(TEN_SECONDS)).until(() -> {
             Request request = new Request.Builder()
```

Comment:
I know it is not scope of this PR, but can you extract this `request` creation out of `await` loop?